Lib/sre_compile.py

```diff
@@ -587,6 +591,9 @@ def _compile_info(code, pattern, flags):
         emit(MAXCODE)
         prefix = prefix[:MAXCODE]
     emit(min(hi, MAXCODE))
+    # REPEAT count
+    assert len(code) == _REPEAT_COUNT_OFFSET
+    emit(0)
```

Comment:
It seems to me that INFO is only used for optimization. If you ignore it, all will work, but maybe slower.

Essential information like the number of groups etc is passed to `_sre.compile()` as separate arguments. The REPEAT count is of such kind. If it is not known we cannot start matching.

I propose to pass it as a separate argument. And in `_validate_inner()` check that all repeat_index are less than that count (like we validate group indices and  the MARK argument).

Reply:
Sounds reasonable, I'll do these.
Please merge your PR first.